### PR TITLE
Use non-terminal extras for comments

### PR DESCRIPTION
**Pull request by 314eter**
I tried to remove some external rules by using tree-sitter/tree-sitter#469.

Since a non-terminal extra is not allowed to appear elsewhere in the grammar, I had to make two identical rules in order to allow nested comments.

There are still some problems:
- ~Escape sequences in string literals in comments are added to the syntax tree, but escape sequences in character literals are not.~
- Comments in characters (`'(* comment *)x'`) are allowed, but shouldn't be. This is because of tree-sitter/tree-sitter#474.

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1532,6 +1549,7 @@ module.exports = grammar({
     string: $ => seq(
       '"',
       repeat(choice(
+        token.immediate('(*'),
```
Comment (314eter) [inline grammar.js]: This is to prevent comments in strings.

File: grammar.js [hunk 2]
```diff
@@ -1543,7 +1561,7 @@ module.exports = grammar({
     ),
 
     escape_sequence: $ => choice(
-      /\\[\\"'ntbr ]/,
+      /\\./,
```
Comment (314eter) [inline grammar.js]: Escape sequences in strings in comments should be more relaxed. It's easier to be relaxed everywhere.

**Comment by 314eter** [comment]
@maxbrunsfeld This PR is ready too. I can't find a way to disallow `'(* comment *)x'`, but allowing too much is not a problem, and no one is going to write that anyway.

I updated the known failures to fix the failing tests.

**Comment by 314eter** [comment]
ping @maxbrunsfeld

**Comment by maxbrunsfeld** [comment]
Sorry for the delay there @314eter; I invited you as a collaborator on the project so that you can just push changes.

**Pull request merged**